### PR TITLE
[CHK-736] Send transaction activated event in azure storage queue

### PR DESCRIPTION
**Pull request by infantesimone**
#### List of Changes

- add activated events in `transactionactivatedevents` queue with visibility timeout set to `paymenToken` timeout;

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by giovanniberti — DISMISSED** [review]

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java [hunk 1]
```diff
@@ -184,14 +201,35 @@ private Mono<TransactionActivatedEvent> newTransactionActivatedEvent(
...
+                            log.debug(
+                                "TransactionActivatedEvent {} expires at {} for transactionId {}",
+                                response.getValue().getMessageId(),
+                                response.getValue().getExpirationTime(),
+                                transactionActivatedEvent.getTransactionId()),
+                        error -> log.error(error.toString()),
+                        () ->
+                            log.debug(
+                                "Complete enqueuing the message TransactionActivatedEvent for transactionId {}!",
+                                transactionActivatedEvent.getTransactionId())));
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java:233]: I'm worried that `doOnNext` may break error chaining from the `sendMessageWithResponse` Mono to the outer Mono, maybe `onErrorResume` may be a better choice of combinator

Comment (infantesimone) [reply src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java:233]: this event is sent if the saving on the eventstore is successful, and then the event on the queue is sent (from this event a "virtual" timer starts to evaluate the pending transaction). I understand your point, so better `doOnSuccess`?

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java [hunk 2]
```diff
@@ -184,14 +218,31 @@ private Mono<TransactionActivatedEvent> newTransactionActivatedEvent(
...
+        .then(Mono.just(transactionActivatedEvent))
+        .onErrorResume(
+            exception -> {
+              log.error(
+                  "Error to generate event TRANSACTION_ACTIVATED_EVENT for rptId {} and transactionId {} - error {}",
+                  transactionActivatedEvent.getRptId(),
+                  transactionActivatedEvent.getTransactionId(),
+                  exception.getMessage());
+              return Mono.error(exception);
+            })
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java:240]:
```suggestion
        .doOnError(
            exception -> {
              log.error(
                  "Error to generate event TRANSACTION_ACTIVATED_EVENT for rptId {} and transactionId {} - error {}",
                  transactionActivatedEvent.getRptId(),
                  transactionActivatedEvent.getTransactionId(),
                  exception.getMessage());
            })
```

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java [hunk 3]
```diff
@@ -14,78 +15,143 @@
...
-						.just(commandData.getData().transactionActivationRequested().getStatus() == TransactionStatusDto.ACTIVATION_REQUESTED))
-				.switchIfEmpty(Mono.error(new AlreadyProcessedException(command.getRptId())))
-				.flatMap(commandData -> {
-					final String paymentToken = commandData.getData().activationResultData().getPaymentToken();
+    return Mono.just(command)
+        .filterWhen(
+            commandData ->
+                Mono.just(
+                    commandData.getData().transactionActivationRequested().getStatus()
+                        == TransactionStatusDto.ACTIVATION_REQUESTED))
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java:59]:
```suggestion
        .filter(
            commandData ->
                commandData.getData().transactionActivationRequested().getStatus()
                    == TransactionStatusDto.ACTIVATION_REQUESTED)
```

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java [hunk 4]
```diff
@@ -14,78 +15,143 @@
...
+                        .then(Mono.just(transactionActivatedEvent))
+                        .onErrorResume(
+                                exception -> {
+                                    log.error(
+                                            "Error to generate event TRANSACTION_ACTIVATED_EVENT for rptId {} and transactionId {} - error {}",
+                                            transactionActivatedEvent.getRptId(),
+                                            transactionActivatedEvent.getTransactionId(),
+                                            exception.getMessage());
+                                    return Mono.error(exception);
+                                })
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateResultHandler.java:148]:
```suggestion
                        .doOnError(
                                exception ->
                                    log.error(
                                            "Error to generate event TRANSACTION_ACTIVATED_EVENT for rptId {} and transactionId {} - error {}",
                                            transactionActivatedEvent.getRptId(),
                                            transactionActivatedEvent.getTransactionId(),
                                            exception.getMessage()))
```

**Review by pietro-tota — DISMISSED** [review]

**Pull request merged**